src/libcrun/linux.c

```diff
@@ -1611,7 +1611,16 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
   int rootfsfd = get_private_data (container)->rootfsfd;
   const char *rootfs = get_private_data (container)->rootfs;
   size_t rootfs_len = get_private_data (container)->rootfs_len;
-  const char *rel_dev = relative_path_under_dev (device->path);
+  // Normalize path by removing trailing slash if it exists.
+  size_t path_len = strlen (device->path);
+  char normalized_path[path_len + 1];
+  snprintf (normalized_path, sizeof (normalized_path), "%s", device->path);
```

Comment:
Since no formatting is needed and we already know the length, then maybe `memcpy` is a better alternative here as compared to `snprintf`.